lib/v8.js

```diff
@@ -247,37 +247,42 @@ Deserializer.prototype.readRawBytes = function readRawBytes(length) {
 
 function arrayBufferViewTypeToIndex(abView) {
   const type = ObjectPrototypeToString(abView);
-  if (type === '[object Int8Array]') return 0;
-  if (type === '[object Uint8Array]') return 1;
-  if (type === '[object Uint8ClampedArray]') return 2;
-  if (type === '[object Int16Array]') return 3;
-  if (type === '[object Uint16Array]') return 4;
-  if (type === '[object Int32Array]') return 5;
-  if (type === '[object Uint32Array]') return 6;
-  if (type === '[object Float32Array]') return 7;
-  if (type === '[object Float64Array]') return 8;
-  if (type === '[object DataView]') return 9;
-  // Index 10 is FastBuffer.
-  if (type === '[object BigInt64Array]') return 11;
-  if (type === '[object BigUint64Array]') return 12;
-  return -1;
+
+  switch (type) {
+    case '[object Int8Array]': return 0;
+    case '[object Uint8Array]': return 1;
+    case '[object Uint8ClampedArray]': return 2;
+    case '[object Int16Array]': return 3;
+    case '[object Uint16Array]': return 4;
+    case '[object Int32Array]': return 5;
+    case '[object Uint32Array]': return 6;
+    case '[object Float32Array]': return 7;
+    case '[object Float64Array]': return 8;
+    case '[object DataView]': return 9;
+      // Index 10 is FastBuffer.
+    case '[object BigInt64Array]': return 11;
+    case '[object BigUint64Array]': return 12;
+    default: return -1;
+  }
 }
 
 function arrayBufferViewIndexToType(index) {
-  if (index === 0) return Int8Array;
-  if (index === 1) return Uint8Array;
-  if (index === 2) return Uint8ClampedArray;
-  if (index === 3) return Int16Array;
-  if (index === 4) return Uint16Array;
-  if (index === 5) return Int32Array;
-  if (index === 6) return Uint32Array;
-  if (index === 7) return Float32Array;
-  if (index === 8) return Float64Array;
-  if (index === 9) return DataView;
-  if (index === 10) return FastBuffer;
-  if (index === 11) return BigInt64Array;
-  if (index === 12) return BigUint64Array;
-  return undefined;
+  switch (index) {
```

Comment:
What about using an array-based lookup table here? That would probably be even more efficient, especially for the last cases.

Reply:
It's faster, I'm updating my changes

```
➜  node git:(perf/v8-deserialize-serialize) ✗ ./node-with-array ./benchmark/v8/serialize.js
v8/serialize.js n=1000000: 637,198.2984177626
➜  node git:(perf/v8-deserialize-serialize) ✗ ./node-perf-v8 ./benchmark/v8/serialize.js
v8/serialize.js n=1000000: 625,952.948029536
➜  node git:(perf/v8-deserialize-serialize) ✗ ./node-with-array ./benchmark/v8/deserialize.js
v8/deserialize.js n=1000000: 1,499,926.0363972564
➜  node git:(perf/v8-deserialize-serialize) ✗ ./node-perf-v8 ./benchmark/v8/deserialize.js
v8/deserialize.js n=1000000: 1,435,339.223199034
```